keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,375 @@
+    string memory_type = 1;
+    uint64 size = 2;
+    TopologyInfo topology = 3;
+}
+
+// ContainerDevices contains information about the devices assigned to a container by device plugin
+message ContainerDevices {
+    string resource_name = 1;
+    repeated string device_ids = 2;
+    TopologyInfo topology = 3;
+}
+
+// Topology describes hardware topology of the resource
+message TopologyInfo {
+    repeated NUMANode nodes = 1;
+}
+
+// NUMA representation of NUMA node
+message NUMANode {
+    int64 ID = 1;
+}
+
+// DynamicResource contains information about the devices assigned to a container by DRA
+message DynamicResource {
+    string class_name = 1;
+    string claim_name = 2;
+    repeated string cdi_devices = 3;
+}
+
+// GetPodResourcesRequest contains information about the pod
+message GetPodResourcesRequest {
+    string name = 1;
+    string namespace = 2;
+}
+
+// GetPodResourcesResponse contains information about the pod the devices
+message GetPodResourcesResponse {
+    PodResources pod_resources = 1;
+}
+
+```
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/apis/podresources`: `01-24-2023` - `61.5%`
+
+##### Integration tests
+
+ These cases will be added in the existing integration tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+##### e2e tests
+
+These cases will be added in the existing e2e tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Feature implemented behind a feature flag.
+- [ ] e2e tests completed and enabled.
+
+#### Beta
+
+- [ ] Gather feedback from consumers of the DRA feature and k8snetworkplumbingwg working group
+- [ ] No major bugs reported in the previous cycle.
+
+#### GA
+
+- [ ] Allowing time for feedback (1 year).
+- [ ] Risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+With gRPC the version is part of the service name.
+Old versions and new versions should always be served and listened by the kubelet.
+
+To a cluster admin upgrading to the newest API version, means upgrading Kubernetes to a newer version as well as upgrading the monitoring component.
+
+To a vendor changes in the API should always be backwards compatible.
+
+### Version Skew Strategy
+
+Kubelet will always be backwards compatible, so going forward existing plugins are not expected to break.
+
+
+## Production Readiness Review Questionnaire
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name for retriving resources allocated by DRA: `DynamicResourceAllocation` and `PodResourcesDynamicResources`.
+  - Feature gate name for Get method: `PodResourcesGet`. In case `DynamicResourceAllocation` and `PodResourcesDynamicResources`
```

Comment:
As I mentioned in my comment here https://github.com/kubernetes/enhancements/pull/3738#discussion_r1096531525, I don't think we should gate the `Get()` call by the `DynamicResourceAllocation` feature gate, only the code relevant for the `PodResourcesDynamicResources` additions.

Looking more closely, it looks like maybe this is a copy-paste error since I assume the `PodResourcesGet` method should actually be gated by the `PodResourcesGet` feature gate and not the `PodResourcesDynamicResources` feature gate anyway.

Reply:
I added the "In case `DynamicResourceAllocation` and `PodResourcesDynamicResources`" part to emphasis that in case
the 3 features gate are enabled PodResourcesGet  , DynamicResourceAllocation  and  PodResourcesDynamicResources the Get call will return the resources allocated by DRA as well. of course If we just enable PodResourcesGet it should return without DRA resources even if we enable  PodResourcesGet  , DynamicResourceAllocation is should return without DRA resources.  Is my understanding correct?